src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx

```diff
@@ -1,53 +1,155 @@
 import { SystemContext } from "Artsy"
 import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
-import React, { useContext } from "react"
-import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import React, { useContext, useState } from "react"
+import {
+  createPaginationContainer,
+  graphql,
+  QueryRenderer,
+  RelayPaginationProp,
+} from "react-relay"
+import styled from "styled-components"
 import { get } from "Utils/get"
+import createLogger from "Utils/logger"
 import { RecommendedArtistFragmentContainer as RecommendedArtist } from "./RecommendedArtist"
 
+const logger = createLogger("ArtistRecommendations.tsx")
+
 interface ArtistRecommendationsProps {
   artist: ArtistRecommendations_artist
+  relay: RelayPaginationProp
 }
 
+const PAGE_SIZE = 3
+
+// TODO:
+// * get help with relay errors
+// * uncomment test & get it working & add coverage for loading more
 export const ArtistRecommendations: React.FC<ArtistRecommendationsProps> = ({
   artist,
+  relay,
 }) => {
+  const [fetchingNextPage, setFetchingNextPage] = useState(false)
+
   const { name } = artist
   const relatedArtists = get(artist, a => a.related.artists.edges, []).map(
-    edge => <RecommendedArtist artist={edge.node} key={edge.node.__id} />
+    edge => <RecommendedArtist artist={edge.node} key={edge.node.id} />
   )
 
+  const fetchData = () => {
+    if (!relay.hasMore() || relay.isLoading()) {
+      return
+    }
+    setFetchingNextPage(true)
+    relay.loadMore(PAGE_SIZE, error => {
+      if (error) {
+        logger.error(error)
+      }
+      setFetchingNextPage(false)
+    })
+  }
+
   return (
     <div>
       <Serif size="8" color="black100">
         Related to {name}
       </Serif>
       {relatedArtists}
+
+      {fetchingNextPage && <FetchingMoreSpinner />}
+
+      {relay.hasMore() && <ShowMoreButton onClick={fetchData} />}
     </div>
   )
 }
 
-export const ArtistRecommendationsFragmentContainer = createFragmentContainer(
+const FetchingMoreSpinner: React.FC = () => {
+  return (
+    <Flex flexDirection="column" alignItems="center">
+      <SpinnerContainer mt={3}>
+        <Spinner />
+      </SpinnerContainer>
+    </Flex>
+  )
+}
+
+const ShowMoreButton: React.FC<{ onClick: () => void }> = ({ onClick }) => {
+  return (
+    <Flex flexDirection="column" alignItems="center">
+      <Button my={4} variant="secondaryOutline" onClick={onClick}>
+        Show More
+      </Button>
+    </Flex>
+  )
+}
+
+const SpinnerContainer = styled(Box)`
+  position: relative;
+`
+
+export const ArtistRecommendationsPaginationContainer = createPaginationContainer(
   ArtistRecommendations,
   {
+    // TODO: can we remove endCursor????
     artist: graphql`
-      fragment ArtistRecommendations_artist on Artist {
+      fragment ArtistRecommendations_artist on Artist
+        @argumentDefinitions(
+          count: { type: "Int", defaultValue: 3 }
```

Comment:
#minor Since we explicitly use `first`/`after`, it might be worth it to just use that everywhere and not use the more generic `count` and `cursor` names

Reply:
Yeah, I think we should probably tackle this all at once across Reaction and Emission. They're functionally the same but both get copied when creating new pagination containers. Maybe we could also deprecate the non-canonical ones in the MP schema?